### PR TITLE
Updated error handling to properly capture undefined, null and NaN ca…

### PR DESCRIPTION
**Pull request by teeehn**
…ses.

### CODE REVIEW

**Comment by teeehn** [comment]
Test here: http://jsfiddle.net/tomnic/ou6jewpc/

File: margincalc.js [hunk 1]
```diff
@@ -10,22 +10,44 @@ margincalc.validate = function (obj) {
     // throws immediately if obj input is not correct
     if (!obj || typeof obj !== 'object') {
         throw new Error(EX_PREFIX + 'Validation input must be an object.');
-        return false;
     }
 
+    // cost must be passed and must not be undefined, null, or NaN
+    if (obj.hasOwnProperty('cost') &&
```
Comment (thisiskeith) [inline margincalc.js]: why hasOwnProperty?

Comment (teeehn) [inline margincalc.js]: The object passed from each method varies in which properties it contains. Using obj.property was proving to be problematic. With text input this is type string and evaluates true, but was expecting false in cases where a number was passed. So I use hasOwnProperty to check for the existence and then check type and value separately.

File: margincalc.js [hunk 2]
```diff
@@ -10,22 +10,35 @@ margincalc.validate = function (obj) {
...
-    }
-    if (obj.revenue && parseFloat(obj.revenue) <= 0) {
-        throw new Error(EX_PREFIX + 'Revenue entered must be greater than 0.');
-        return false;
+    var config = [
+        {name: 'cost', rule: 1, label: 'Cost'},
+        {name: 'revenue', rule: 1, label: 'Revenue'},
+        {name: 'grossMargin', rule: 2, label: 'Gross margin'}];
+
+    for (var i in config) {
```
Comment (thisiskeith) [inline margincalc.js]: never declare a variable in a loop. 

``` js
var i;
for (i in config) {
```

**Pull request merged**